plugins/search/src/components/SearchResult/SearchResult.tsx

```diff
@@ -0,0 +1,230 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useState, useEffect } from 'react';
+import { useAsync } from 'react-use';
+
+import { makeStyles, Typography, Grid, Divider } from '@material-ui/core';
+import { Table, TableColumn } from '@backstage/core';
+import { FilterButton, Filters } from '../Filters';
+
+import SearchApi from '../../apis';
+
+const useStyles = makeStyles(() => ({
+  searchTerm: {
+    background: '#eee',
+    padding: '0.2%',
+    borderRadius: '10%',
+  },
+  tableHeader: {
+    marginTop: '10px',
+    display: 'flex',
+  },
+  divider: {
+    width: '1px',
+    margin: '0 20px',
+    padding: '15px 0',
+  },
+}));
+
+type SearchResultProps = {
+  currentTarget?: string;
+};
+
+type TableHeaderProps = {
+  searchTerm?: string;
+  numberOfSelectedFilters: number;
+  numberOfResults: number;
+  handleToggleFilters: () => any;
+};
+
+type Filters = {
+  selected: string;
+  checked: Array<string | null>;
+};
+
+// TODO: move out column to make the search result component more generic
+const columns: TableColumn[] = [
+  {
+    title: 'Component Id',
+    field: 'name',
+    highlight: true,
+  },
+  {
+    title: 'Description',
+    field: 'description',
+  },
+  {
+    title: 'Owner',
+    field: 'owner',
+  },
+  {
+    title: 'Kind',
+    field: 'kind',
+  },
+  {
+    title: 'LifeCycle',
+    field: 'lifecycle',
+  },
+];
+
+const TableHeader = ({
+  searchTerm,
+  numberOfSelectedFilters,
+  numberOfResults,
+  handleToggleFilters,
+}: TableHeaderProps) => {
+  const classes = useStyles();
+
+  return (
+    <div className={classes.tableHeader}>
+      <FilterButton
+        numberOfSelectedFilters={numberOfSelectedFilters}
+        handleToggleFilters={handleToggleFilters}
+      />
+      <Divider className={classes.divider} orientation="vertical" />
+      <Grid item sm={12}>
+        {searchTerm ? (
+          <Typography variant="h6">
+            {`${numberOfResults} `}
+            {numberOfResults > 1 ? `results for ` : `result for `}
+            <span className={classes.searchTerm}>"{searchTerm}"</span>{' '}
+          </Typography>
+        ) : (
+          <Typography variant="h6">{`${numberOfResults} results`}</Typography>
+        )}
+      </Grid>
+    </div>
+  );
+};
+
+const SearchResult = ({ currentTarget }: SearchResultProps) => {
+  const [showFilters, toggleFilters] = useState(false);
+  const [filters, setFilters] = useState<Filters>({
+    selected: 'All',
+    checked: [],
+  });
+
+  const [filteredResults, setFilteredResults] = useState<Array<object>>([]);
+
+  const searchApi = new SearchApi();
+
+  const { loading, error, value: results } = useAsync(() => {
+    return searchApi.getSearchData();
+  }, []);
+
+  useEffect(() => {
+    if (results) {
+      let withFilters = results;
+
+      // apply filters
+
+      // filter on selected
+      if (filters.selected !== 'All') {
```

Comment:
any particular reason this one was named `selected` instead of `kind`? same for `checked` below

Reply:
the idea from the beginning was to keep the search result page generic to what ever structure the search result data have and therefore also the filters, that was the only reason really but how it is now I think I could have just make it more clear by using kind and lifecycle as it is. We will most likely rethink the search result component soon, to avoid use just another table, so lets change this when we do that